sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java

```diff
@@ -462,9 +452,7 @@ private IAccount retrieveAccount(String id) {
     @WorkerThread
     private synchronized void selectSignInTypeAndSignIn() {
         if (mAuthenticationClient == null) {
-            AppCenterLog.debug(LOG_TAG, "signIn is called while it's not configured, waiting.");
-            mSignInDelayed = true;
-            return;
+            completeSignIn(null, new IOException("signIn is called while it's not configured, waiting."));
```

Comment:
This seems to be the correct type of exception, I'm just surprised it's not something more descriptive like `NetworkException`.  Nothing to change here, just documenting my thoughts :)